NewForce-Cohort-7/tabloidmvc-lone-star-cowboys
# Jf create category

**Description**
This pull request is for TabloidMVC Trello ticket #10 – Create a Category - https://trello.com/c/t192cOfX/10-10-create-a-category . The functionality of adding a new category has been added. A user who is logged in and selects the Category Management link in the navbar should be directed to a page that lists all Category names alphabetically. When the user clicks the Create New button on this page, a form should appear where they can enter their category name. After clicking save, the category should be saved to the database and redirected to the Category list page.

The branch for the pull request: jf-createCategory

Updates:
Added "Category Management " to the nav bar - Views - Shared - _Layout.cshtml
Category Repository – Added Method
Category Control – Edited Method
Category View – Created (Create.cshtml file) 

Type of change
[ x ] New feature (non-breaking change which adds functionality)


Testing Instructions:
Testing Instructions
Follow these instructions to protect the branch you are working on and to test this pull request: https://gist.github.com/jordan-castelloe/96f76d5bd4cfbd2181084af674c2a1af#when-you-need-to-test-a-teammates-pr
If you do not currently have the code for this project, clone the repository - https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys.
Open the TabloidMVC.sln in visual studio
Prepare your database.
•	In the github repository: https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys
•	Open the SQL folder: https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/tree/main/SQL
•	there are two files - [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/blob/main/SQL/01_TabloidMVC_Create_DB.sql) and [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)
•	In Visual Studio in the top navbar click View and select SQL Server Object Explorer. Open the server marked as "SQLEXPRESS" in the SQL Server Object Explorer on the left side of Visual Code. Once there, add a new SQL server by selecting the 'Add SQL Server' option at the top of the side bar. Copy the data from the [seed data](https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/tree/main/SQL) and execute by pressing the green Play button.
•	start the server by clicking the solid green play button in the middle top of your screen. A browser window will display with the Tabloid landing page.
•	In the nav bar, click Login. You only need an email to login use [admin@example.com](mailto:admin@example.com)
•	You now see the home page for the site. In the navbar, click on "Category Management. "
Once on this page, you will see a list of categories including: 

Close Magic 

Cooking Cthulhu 

Sightings 

Dance 

Dance 

Entertainment 

History

 Home and Garden

 Improv 

Movies 

Music

 Politics

 Regrets

When you click on the 'Create New' link, you'll be taken to a form where you can create new categories: In the create new category box, type "Food" then click "Create". You should be redirected to the category list page and see your new category "Food" under the word "Entertainment" and above the word "History".
